src/stub/preprocessor/forward_declarations.rs

```diff
@@ -0,0 +1,120 @@
+use super::Renderable;
+use crate::stub::{Cmd, Stub, VarType, VariableCommand};
+
+const ALPHABET: [char; 18] = [
+    'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+];
```

Comment:
I don't like that this is completely independent of the corresponding `ALPHABET` in renderer.rs. Whoever modifies one of these now also needs to remember to change the other one.

I think it would be good to at least have a comment documenting this indirect dependency above both const declarations. Even better would be to find a way to use the same data in both, for example by storing the loop variable in the `Loop` / `Loopline` Cmd so the preprocessor can just look at that instead of keeping track of nesting depth itself.

Reply:
I will make the ALPHABET public in renderer where it belongs. This should still not leak to the API since `renderer.rs` is private in `stub.rs`.